File: Workspace/SPOJ/Adding_Reversed_Number.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int x,y,t;
    cin>>t;
    while(t--)
    {
        int a=0,b=0;
        cin>>x>>y;
        while(x!=0)
        {
            a=a*10+(x%10);
            x/=10;
        }
        // cout<<a<< " ";
        while(y!=0)
        {
            b=b*10+(y%10);
            y/=10;
        }
        //cout<<b<<" ";
        int sum=a+b;
        //cout<<sum<<endl;
        int s=0;
        while(sum!=0)
        {
            s=(s*10)+(sum%10);
            sum/=10;
        }
        cout<<s<<endl;
    }
}

File: Workspace/Mostofa_Saad_Ibrahim_With_Abdallah/PS/src/Adding_Reversed_Number.java
// 27-03-20
import java.math.BigInteger;
import java.util.Scanner;

public class Adding_Reversed_Number {
    public static void main(String []args){

        int tc;
        BigInteger a,b,sum;
        StringBuffer sb;

        Scanner cin = new Scanner(System.in);
        tc = cin.nextInt();

        while(tc>0){

            sb = new StringBuffer(cin.next());
            sb = sb.reverse();
            a = new BigInteger(sb.toString());
            sb = new StringBuffer(cin.next());
            sb = sb.reverse();
            b = new BigInteger(sb.toString());

            sum = a.add(b);
            sb = new StringBuffer(sum.toString());
            sb = sb.reverse();
            sum = new BigInteger(sb.toString());

            System.out.println(sum);
            tc--;
       }
    }
}
